src/net.cpp

```diff
@@ -2396,6 +2399,63 @@ bool CConnman::MaybePickPreferredNetwork(std::optional<Network>& network)
     return false;
 }
 
+/**
+ * Decide whether to open a private broadcast connection and if yes, to which network.
+ * @param[in,out] net If the previous attempt to open a connection was an attempt to
+ * open a private broadcast connection (successful or not), then this should have a
+ * value. When the function ends if this has value then it will contain the network
+ * to which to open the connection.
+ * @param[in] num_needed Number of private broadcast connections that need to be opened.
+ * @param[in] num_opened Number of private broadcast connections that are currently opened.
+ */
+static void MaybePickPrivateBroadcastNetwork(std::optional<Network>& net, size_t num_needed, size_t num_opened)
```

Comment:
In b1f65ba837a805dc8f855c02337c1593c77add6d: non-blocker: I think it would be more elegant if `MaybePickPrivateBroadcastNetwork` returns `std::optional<Network> net`.

Reply:
Done.